x-pack/plugins/observability_solution/apm_data_access/server/services/get_host_names/index.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { estypes } from '@elastic/elasticsearch';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+import { RegisterParams } from '../register_services';
+
+export interface ServicesHostNamesParams {
+  query: estypes.QueryDslQueryContainer;
+  from: number;
+  to: number;
+  limit: number;
+}
+
+export function createGetHostNames(params: RegisterParams) {
+  return async ({ from: timeFrom, to: timeTo, query, limit }: ServicesHostNamesParams) => {
+    const { apmIndices, esClient } = await params.getResourcesForServices();
+
+    const esResponse = await esClient.search({
+      index: [apmIndices.metric, apmIndices.transaction, apmIndices.span, apmIndices.error],
+      track_total_hits: false,
+      ignore_unavailable: true,
+      size: 0,
+      query: {
+        bool: {
+          filter: [query, ...rangeQuery(timeFrom, timeTo)],
+        },
+      },
+      aggs: {
+        hostNames: {
+          terms: {
+            field: 'host.name',
```

Comment:
there is a constant for this somewhere (`HOST_NAME`).

Reply:
not in the `apm_data_access`. we need to either move the constants from `apm` plugin here or create a new constants file.

Reply:
yeah, let's move the constants then

Reply:
I'll do that in a follow up PR, over 250 files will be changed after this

Reply:
package might be better at that point, WDYT?

Reply:
yeah. When doing this https://github.com/elastic/kibana/pull/189046#discussion_r1694931076, some types could also be moved to this new package.